Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/qrcode/consent/SubmissionConsentFragment.kt

```diff
@@ -23,10 +33,14 @@ class SubmissionConsentFragment : Fragment(R.layout.fragment_submission_consent)
     @Inject lateinit var viewModelFactory: CWAViewModelFactoryProvider.Factory
     private val viewModel: SubmissionConsentViewModel by cwaViewModels { viewModelFactory }
     private val binding: FragmentSubmissionConsentBinding by viewBindingLazy()
+    private val navArgs by navArgs<SubmissionConsentFragmentArgs>()
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
         binding.viewModel = viewModel
+        navArgs.qrCode?.let {
+            viewModel.qrCode = it
+        }
```

Comment:
how do you feel about passing this the viewModel as `Assisted` parameter ?

Reply:
Personal opinion: Too complicated for such a simple task